---
title: "Detect batch effect - report 1"
author: "Qunhua Li"
date: "7/12/2017"
output: html_document
---
# R Markdown
First, one must load the data and the packages need to analyze it.

```{r load packages, message=FALSE, warning=FALSE}
# source("http://bioconductor.org/biocLite.R")
# biocLite("Biobase")
# biocLite("genefilter")
# devtools::install_github("genomicsclass/GSE5859")

library(Biobase)
library(rafalib)
library(GSE5859)
library(genefilter)

data(GSE5859)
```

Also, you need to extract the expression data into a usable (matrix) format.

```{r }
# gene expression levels  
geneExpression = exprs(e)
```


Now we can check years that the data from different ethnic groups were collected. As well how the date relates to the ethnicity.

# Data collection dates
```{r }
# print date info
sampleInfo = pData(e)
head(sampleInfo$date)
```

```{r }
# Print a table of year vs ethnic groups
year = factor( format(sampleInfo$date,"%y") )
tab = table(year,sampleInfo$ethnicity)
print(tab)
```

In order to compare experimental data, we needt to extract the genes of interest.
# Differential expression analysis
```{r }
##remove control genes
out <- grep("AFFX",rownames(geneExpression))
```

We first use a t-test to perform differential expression analysis between CEU and ASN.
```{r }
eth <- sampleInfo$ethnicity
ind<- which(eth%in%c("CEU","ASN"))
res1 <- rowttests(geneExpression[-out,ind],droplevels(eth[ind]))
```

We then perform the same analysis between 2002 vs 2003 within CEU.
```{r }
ind <- which(year%in%c("02","03") & eth=="CEU")
res2 <- rowttests(geneExpression[-out,ind],droplevels(year[ind]))
```

We plot histograms for p-values as in Akey et al (2007). 
```{r }   
mypar(2, 2)
hist(res1$p, main="p-value of CEU vs ASN")
hist(res2$p, main="p-value of 2002 vs 2003 for CEU")
```

There are a lot of significant genes between the CEU samples collected in 2002 and 2003. 